### DISCUSSION TITLE
How to transform a single query output (react-query).

### DISCUSSION THREAD

**Discussion by oaltun** [orval-labs/orval#746] [Q&A]
Hi,

Orval generated a useSomeResourceList hook for me. I can use it and get a list of the resource. The list needs to be filtered/transformed before I use it. 

React-query allows such transformation using a select parameter:

```
export const useTodosQuery = () =>
  useQuery({
    queryKey: ['todos'],
    queryFn: fetchTodos,
    select: (data) => data.map((todo) => todo.name.toUpperCase()),
  })
```

How can we accomplish that in Orval generated hooks?

**Comment by gegorov** [comment]
i think you can do it now,  your useSomeResourceList accepts two parameters. second parameter is options which has query property. this query property is `Partial<UseQueryOptions<T>>`

so you can make something like this 

`const { data: transformedData } = useSomeResourceList(params, { query: { select: (data) => data.map((todo) => todo.name.toUpperCase()) }})`
